[Bug] ctypes: AttributeError: module 'collections' has no attribute 'Callable' (GRASS GIS 7.8.5)

**Describe the bug**
A clear and concise description of what the bug is.

While trying to address the current Fedora packaging issue (https://bugzilla.redhat.com/show_bug.cgi?id=1973621) I identified the following problem: 

With Python 3.10 the ctypes compilation fails:

```
GISRC=/builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu/demolocation/.grassrc78 GISBASE=/builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu PATH="/builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu/bin:/builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu/bin:/builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu/scripts:$PATH" PYTHONPATH="/builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu/etc/python:/builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu/gui/wxpython:$PYTHONPATH" LD_LIBRARY_PATH="/builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu/bin:/builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu/bin:/builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu/scripts:/builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu/lib:/builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu/lib:" LC_ALL=C LANG=C LANGUAGE=C ./ctypesgen.py --cpp "gcc -E   ""     -I/builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu/include -I/builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu/include -D__GLIBC_HAVE_LONG_LONG" -lgrass_imagery.7.8   /builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu/include/grass/imagery.h /builddir/build/BUILD/grass-7.8.5/dist.x86_64-redhat-linux-gnu/include/grass/defs/imagery.h -o OBJ.x86_64-redhat-linux-gnu/imagery.py
Traceback (most recent call last):
Traceback (most recent call last):
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/./ctypesgen.py", line 162, in <module>
    descriptions = ctypesgencore.parser.parse(options.headers, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/__init__.py", line 21, in parse
    parser = DataCollectingParser(headers, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/datacollectingparser.py", line 31, in __init__
    ctypesparser.CtypesParser.__init__(self, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/ctypesparser.py", line 63, in __init__
    super(CtypesParser, self).__init__(options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/cparser.py", line 90, in __init__
    self.preprocessor_parser = preprocessor.PreprocessorParser(options, self)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/preprocessor.py", line 137, in __init__
    self.lexer = lex.lex(cls=PreprocessorLexer,
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/lex.py", line 723, in lex
    if isinstance(t, collections.Callable):
AttributeError: module 'collections' has no attribute 'Callable'
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/./ctypesgen.py", line 162, in <module>
    descriptions = ctypesgencore.parser.parse(options.headers, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/__init__.py", line 21, in parse
    parser = DataCollectingParser(headers, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/datacollectingparser.py", line 31, in __init__
    ctypesparser.CtypesParser.__init__(self, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/ctypesparser.py", line 63, in __init__
    super(CtypesParser, self).__init__(options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/cparser.py", line 90, in __init__
    self.preprocessor_parser = preprocessor.PreprocessorParser(options, self)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/preprocessor.py", line 137, in __init__
    self.lexer = lex.lex(cls=PreprocessorLexer,
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/lex.py", line 723, in lex
    if isinstance(t, collections.Callable):
AttributeError: module 'collections' has no attribute 'Callable'
Traceback (most recent call last):
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/./ctypesgen.py", line 162, in <module>
    descriptions = ctypesgencore.parser.parse(options.headers, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/__init__.py", line 21, in parse
    parser = DataCollectingParser(headers, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/datacollectingparser.py", line 31, in __init__
    ctypesparser.CtypesParser.__init__(self, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/ctypesparser.py", line 63, in __init__
    super(CtypesParser, self).__init__(options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/cparser.py", line 90, in __init__
    self.preprocessor_parser = preprocessor.PreprocessorParser(options, self)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/preprocessor.py", line 137, in __init__
    self.lexer = lex.lex(cls=PreprocessorLexer,
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/lex.py", line 723, in lex
    if isinstance(t, collections.Callable):
AttributeError: module 'collections' has no attribute 'Callable'
Traceback (most recent call last):
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/./ctypesgen.py", line 162, in <module>
    descriptions = ctypesgencore.parser.parse(options.headers, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/__init__.py", line 21, in parse
    parser = DataCollectingParser(headers, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/datacollectingparser.py", line 31, in __init__
    ctypesparser.CtypesParser.__init__(self, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/ctypesparser.py", line 63, in __init__
make[6]: *** [Makefile:107: OBJ.x86_64-redhat-linux-gnu/gmath.py] Error 1
make[6]: *** Waiting for unfinished jobs....
    super(CtypesParser, self).__init__(options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/cparser.py", line 90, in __init__
    self.preprocessor_parser = preprocessor.PreprocessorParser(options, self)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/preprocessor.py", line 137, in __init__
Traceback (most recent call last):
    self.lexer = lex.lex(cls=PreprocessorLexer,
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/lex.py", line 723, in lex
    if isinstance(t, collections.Callable):
AttributeError: module 'collections' has no attribute 'Callable'
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/./ctypesgen.py", line 162, in <module>
    descriptions = ctypesgencore.parser.parse(options.headers, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/__init__.py", line 21, in parse
make[6]: *** [Makefile:107: OBJ.x86_64-redhat-linux-gnu/proj.py] Error 1
    parser = DataCollectingParser(headers, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/datacollectingparser.py", line 31, in __init__
    ctypesparser.CtypesParser.__init__(self, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/ctypesparser.py", line 63, in __init__
    super(CtypesParser, self).__init__(options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/cparser.py", line 90, in __init__
    self.preprocessor_parser = preprocessor.PreprocessorParser(options, self)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/preprocessor.py", line 137, in __init__
    self.lexer = lex.lex(cls=PreprocessorLexer,
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/lex.py", line 723, in lex
    if isinstance(t, collections.Callable):
AttributeError: module 'collections' has no attribute 'Callable'
make[6]: *** [Makefile:107: OBJ.x86_64-redhat-linux-gnu/date.py] Error 1
make[6]: *** [Makefile:107: OBJ.x86_64-redhat-linux-gnu/gis.py] Error 1
make[6]: *** [Makefile:107: OBJ.x86_64-redhat-linux-gnu/raster.py] Error 1
Traceback (most recent call last):
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/./ctypesgen.py", line 162, in <module>
    descriptions = ctypesgencore.parser.parse(options.headers, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/__init__.py", line 21, in parse
    parser = DataCollectingParser(headers, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/datacollectingparser.py", line 31, in __init__
    ctypesparser.CtypesParser.__init__(self, options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/ctypesparser.py", line 63, in __init__
    super(CtypesParser, self).__init__(options)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/cparser.py", line 90, in __init__
    self.preprocessor_parser = preprocessor.PreprocessorParser(options, self)
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/preprocessor.py", line 137, in __init__
    self.lexer = lex.lex(cls=PreprocessorLexer,
  File "/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/lex.py", line 723, in lex
    if isinstance(t, collections.Callable):
AttributeError: module 'collections' has no attribute 'Callable'
make[6]: Leaving directory '/builddir/build/BUILD/grass-7.8.5/lib/python/ctypes'
make[6]: *** [Makefile:107: OBJ.x86_64-redhat-linux-gnu/imagery.py] Error 1
make[5]: *** [Makefile:86: default] Error 2
```

**To Reproduce**

The issue happens on the Fedora build system (likely on others as well once Python 3.10 is more widely in use):

- https://kojipkgs.fedoraproject.org//work/tasks/9252/70549252/build.log
- https://kojipkgs.fedoraproject.org//work/tasks/9252/70549252/root.log (packages)

**System description (please complete the following information):**

- Operating System: Fedora 35
- GRASS GIS version 7.8.5
- python 3.10.0~b3-1.fc35

It would be ideal to get this fixed before the G 7.8.6 release.

--- a/python/grass/ctypes/ctypesgencore/parser/lex.py
+++ b/python/grass/ctypes/ctypesgencore/parser/lex.py
@@ -348,7 +348,7 @@ class Lexer:
                     break
 
                 # if func not callable, it means it's an ignored token
-                if not isinstance(func, collections.Callable):
+                if not isinstance(func, collections.abc.Callable):
                     break
 
                 # If token is processed by a function, call it
@@ -720,7 +720,7 @@ def lex(module=None, object=None, debug=0, optimize=0,
         states, tokname = _statetoken(f, stateinfo)
         toknames[f] = tokname
 
-        if isinstance(t, collections.Callable):
+        if isinstance(t, collections.abc.Callable):
             for s in states:
                 funcsym[s].append((f, t))
         elif (isinstance(t, bytes) or isinstance(t, str)):

